app/controllers/flags_controller.rb

```diff
@@ -145,16 +145,26 @@ def create
       redirect_to root_path
     end
 
-    if @flag = Flag.where(create_options).where(resolved: true).first
+    @create_comment = false
+    if @flag = Flag.where( create_options.except( "initial_comment_body" ) ).where( resolved: true ).first
       @flag.resolved = false
     else
       @flag = @object.flags.build(create_options)
+      @create_comment = create_options.key?( :initial_comment_body ) && create_options[:initial_comment_body].length > 0
+      if @create_comment
+        @comment_attributes = { parent: @flag, user: current_user, body: create_options[:initial_comment_body] }
+        @flag.comments.build( @comment_attributes )
+      end
     end
     if @flag.flag == "other" && !params[:flag_explanation].blank?
       @flag.flag = params[:flag_explanation]
     end
+
     if @flag.save
-      flash[:notice] = t(:flag_saved_thanks_html, url: url_for( @flag ) )
+      flash[:notice] = t( :flag_saved_thanks_html, url: url_for( @flag ) )
+      if @create_comment && Comment.where( @comment_attributes ).length == 0
```

Comment:
Minor suggestion - this query works fine, but I wonder if it would be more readable and succinct to use the model you just created - i.e. `if @create_comment && !@flag.comments.any?`. Not necessary, just offering it as a possibility

Reply:
Your suggested edit doesn't get the trick done, so I've left it as before.